lib/engine-jit/src/artifact.rs

```diff
@@ -141,21 +137,26 @@ impl JITArtifact {
     }
 
     /// Deserialize a JITArtifact
-    pub fn deserialize(jit: &JITEngine, bytes: &[u8]) -> Result<Self, DeserializeError> {
+    ///
+    /// # Safety
+    /// This function is unsafe because rkyv reads directly without validating
+    /// the data.
```

Comment:
Minor nit, but it's useful (at least to me) to also phrase it in a way that is proactive for the prospective caller of the function. i.e. I'm writing some code or reviewing some code and I want to call this function / see if a call to this function is correct; what invariant should I be looking for to check that this function is being called correctly? The answer is probably that the serialized data came from a known valid source.

What's our threat model for maliciously constructed serialization data? Perhaps we need a signature system to mitigate some of these threats. Or at least a hash to fight corruption. Especially if corruption will lead to UB in our system

Either way, I think this is a potentially interesting part of the system from a security / product perspective so it'd be good to think this through and be explicit about what we expect in the safety comments.

Reply:
If you do need validation capabilities, rkyv supports archive validation with [`check_archived_value`](https://docs.rs/rkyv/0.6.1/rkyv/validation/fn.check_archived_value.html). You'll incur a performance hit (see [benches](https://github.com/djkoloski/rust_serialization_benchmark) for details) but it's optional so you could put it behind a command line argument.

Reply:
Fixed!

Reply:
I tried to add optional validation, but the scope was bigger than I envisioned (bytecheck dependency, adding an attr to almost all structs, ...), so it will be better to do it in a PR apart